### ISSUE TITLE
STM32F746 Discovery Board Goes to Safe Mode

### ISSUE THREAD

**Issue by pataelmo** [adafruit/circuitpython#8402]
### CircuitPython version

```python
Adafruit CircuitPython 8.2.6 on 2023-09-12; ST STM32F746G Discovery with STM32F746
Board ID:stm32f746g_discovery
```

### Code/REPL

```python
print("Hello World")
```

### Behavior

I programmed the "adafruit-circuitpython-stm32f746g_discovery-en_US-8.2.6.bin" file from the official repository and tried to connect to the REPL.  I tried just to see what was available but running "import board" printed a few extra new lines and then hung.

Either pressing Ctrl-C or Ctrl-D or saving the code.py file to the drive got me the following errors on the REPL port:
Auto-reload is off.

Running in safe mode! Not running saved code.

You are in safe mode because:
CircuitPython core code crashed hard. Whoops!
NLR jump failed. Likely memory corruption.
Please file an issue with your program at github.com/adafruit/circuitpython/issues.
Press reset to exit safe mode.

Press any key to enter the REPL. Use CTRL-D to reload.

### Description

I programmed the bin using the embedded ST-Link and STM32Cube programmer on windows.  Apologies if I'm making a noob mistake on how to correctly program the board

**Comment by tannewt** [comment]
I don't think we have may folks using CircuitPython on the F7 and this is likely a bug. (Us Adafruit-funded folks haven't spent time on it since we don't have any F7 boards of our own.)

We'd appreciate the help debugging it if you are interested. We can answer dev questions [on Discord](https://adafru.it/discord) in the #circuitpython-dev channel. Thanks!
